gms/gossiper.cc

```diff
@@ -654,6 +666,17 @@ future<> gossiper::apply_state_locally(std::map<inet_address, endpoint_state> ma
 
     logger.debug("apply_state_locally() took {} ms", std::chrono::duration_cast<std::chrono::milliseconds>(
         std::chrono::steady_clock::now() - start).count());
+  }
+}
+
+future<> gossiper::apply_state_locally(std::map<inet_address, endpoint_state> map) {
+    for (auto& entry : map) {
+        //FIXME: this probably has to be much more sophisticated than this, checking generations, etc.
+        //Q: are state updates always full overwrites? Do we ever have to do some merging?
+        _staging_endpoint_state_map[entry.first] = std::move(entry.second);
```

Comment:
The endpoint_state to be applied is only partial. So the code here will overwrite any previous collected states.

Reply:
The merge here is actually going to be pretty complicated. We need to consider both version and generations. This is going to replicate lots of the actual apply logic.